main.tf

```diff
@@ -80,6 +81,15 @@ resource "aws_s3_bucket" "default" {
     }
   }
 
+  dynamic logging {
+    for_each = local.logging_configuration
+
+    content {
+      target_bucket = local.logging_configuration.target_bucket
+      target_prefix = local.logging_configuration.target_prefix
+    }
+  }
```

Comment:
```suggestion
  dynamic logging {
    for_each = local.logging

    content {
      target_bucket = var.logging.target_bucket
      target_prefix = var.logging.target_prefix
    }
  }
```

Reply:
When you don't want to set a prefix, this can be used with an object like:

```
...
logging = {
    target_bucket = "something"
    target_prefix = null
}
...
```